### PR TITLE
Demo branch2

### PR DESCRIPTION
**Pull request by mell0kat**
Important things to note:
- Implement local auth
- used modals for signup/login
  *created test environment && added test script

### CODE REVIEW

File: browser/js/signup/signup.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+             <p ng-show="signupForm.animalName.$invalid &&!signupForm.animalName.$pristine" class="help-block">A name for your animal is required (even if it is silly as could be).</p>
+        </div>
+        <div class="form-group" ng-class= "{'has-error' : signupForm.password.$invalid && !signupForm.password.$pristine} ">
+            <label>Password</label>
+            <input type="password" name="password" ng-model="signup.password" class="form-control" placeholder="'Caput Draconis,' maybe?" required>
+             <p ng-show="signupForm.password.$invalid &&!signupForm.password.$pristine" class="help-block">A password is required (make it #unbreakable).</p>
+        </div>
+        <button type="submit" class="btn btn-default" ng-disabled="signupForm.$invalid">Submit</button> <a ng-click="login()">already have a demo account? </a>
+    </form>
+</div>
```
Comment (wnmprk) [inline browser/js/signup/signup.html:20]: is this diff from the firsttimeuser state?

File: browser/scss/demo/main.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#demo-modal {
+	background: hsl(218, 80%, 94%)  ;
+
+	border: solid 0.5em hsl(218, 80%, 30%);
```
Comment (wnmprk) [inline browser/scss/demo/main.scss:4]: woahhh crazy css. never seen this before with border

File: package.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
   "main": "server/start.js",
   "scripts": {
     "start": "nodemon --watch server -e js,html server/start.js",
-    "postinstall": "gulp build"
+    "postinstall": "gulp build",
+    "test-server": "NODE_ENV=test mocha --watch --reporter spec \"./tests/server/**/*.spec.js\""
```
Comment (wnmprk) [inline package.json:9]: TDD FTW

File: server/db/models/user.js [hunk 4]
```diff
@@ -107,6 +110,26 @@ var schema = new Schema({
...
+            default: 600
+        },
+        weekSteps: [{
+            steps: Number,
+            date: String
+        }],
+        weekSleep: [{
+            minutes: Number,
+            date: String
+        }]
```
Comment (wnmprk) [inline server/db/models/user.js:73]: are we gonna have fake data for demo users?

File: browser/js/signup/signup.html (hunk 1 above)
Comment (rpastor90) [inline browser/js/signup/signup.html:20]: I think this is a modal??

File: seed.js [hunk 5]
```diff
@@ -253,6 +264,13 @@ var seedSwag = function () {
 }; 
 
 connectToDb.then(function () {
+    User.findAsync({}).then(function (users) {
+        if (users.length === 0) {
+            return seedUser();
+        } else {
+            console.log(chalk.magenta('Users have been seeded!'));
+        }
+    })
```
Comment (rpastor90) [inline seed.js]: @mell0kat changed this locally, I think it'll get fixed in an updated PR

**Pull request merged**